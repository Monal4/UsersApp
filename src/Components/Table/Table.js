import React from 'react'
import { useHistory } from 'react-router-dom'
import './table.css'

const Table = ({headers, body}) => {

    const history = useHistory();

    return(
        <table className = "table">
            <thead className="thead">
                <tr>
                    {headers.map(header => 
                        <th key={header}> {header} </th>    
                    )}
                </tr>
            </thead>
            <tbody>
                {body}
            </tbody>
        </table>
    )
}

export default Table;