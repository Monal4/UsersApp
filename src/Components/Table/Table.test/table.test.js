import React from 'react';
import { shallow } from "enzyme";
import '../../../setup';
import Table from "../Table";

describe('Table Component', () => {
    
    it('renders props', () => {
        expect(shallow(<Table headers={["Test"]} body={"Test"}/>).length).toBe(1);
    });

    it('should render snapshot', () => {
        expect(shallow(<Table headers={["Test"]} body={"Test"}/>)).toMatchSnapshot();
    });

    it('validates table headers and body', () => {
        const headers = ["TestHeader1","TestHeader2"];
        const body = ["TestBody2", "TestBody2"];

        const container = shallow(<Table headers={headers} body={body}/>);

        const table = container.find('table');

        const thead = table.find('thead');
        expect(thead).toHaveLength(1);

        const Tableheaders = thead.find('th');
        expect(Tableheaders).toHaveLength(headers.length);

        const tbody = table.find('tbody');
        expect(tbody).toHaveLength(1);
    })
});

