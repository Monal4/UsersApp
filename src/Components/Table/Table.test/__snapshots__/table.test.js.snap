// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component should render snapshot 1`] = `
<table
  className="table"
>
  <thead
    className="thead"
  >
    <tr>
      <th>
         
        Test
         
      </th>
    </tr>
  </thead>
  <tbody>
    Test
  </tbody>
</table>
`;
